dask/array/tests/test_routines.py

```diff
@@ -104,6 +104,36 @@ def test_dot_method():
     assert_eq(a.dot(b), x.dot(y))
 
 
+@pytest.mark.parametrize('func1d_name, func1d', [
+    ["ndim", lambda x: x.ndim],
+    ["sum", lambda x: x.sum()],
+    ["range", lambda x: [x.min(), x.max()]],
+    ["range2", lambda x: [[x.min()], [x.max()]]],
+])
+@pytest.mark.parametrize('shape, axis', [
+    [(10, 15, 20), 0],
+    [(10, 15, 20), 1],
+    [(10, 15, 20), 2],
+    [(10, 15, 20), -1],
+])
+def test_apply_along_axis(func1d_name, func1d, shape, axis):
+    if (func1d_name == "range2" and
+            LooseVersion(np.__version__) < LooseVersion("1.13.0")):
+            pytest.skip(
+                "NumPy %s doesn't support multidimensional results with"
+                " `apply_along_axis`. Need NumPy 1.13.0 or greater."
+                "" % np.__version__
```

Comment:
As a ref, please see NumPy 1.13.0's [release notes]( https://github.com/numpy/numpy/blob/v1.13.0/doc/release/1.13.0-notes.rst#support-for-returning-arrays-of-arbitrary-dimensions-in-apply_along_axis ), which mentions `apply_along_axis` getting support for `func1d`'s that return N-D arrays (with N > 2).